packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2940,7 +2947,8 @@ class _RouteEntry extends RouteTransitionRecord {
   /// remove as a result of a page update.
   static const int kDebugPopAttemptLimit = 100;
 
-  static final Route<dynamic> notAnnounced = _NotAnnounced();
+  static Route<dynamic> get notAnnounced => _notAnnounced ??= _NotAnnounced();
+  static Route<dynamic>? _notAnnounced;
```

Comment:
this is literally a sentinel value, it should definitely not be able to change identity. It's kind of weird that it has state. I would have expected to use `null` here.